### PR TITLE
Firestore API

### PR DESCRIPTION
**Pull request by hiranya911**
Usage:

```
from firebase_admin import firestore

db = firestore.client()
```

### CODE REVIEW

**Comment by hiranya911** [comment]
The build failure is caused by https://github.com/GoogleCloudPlatform/google-cloud-python/issues/4115

Awaiting resolution. In the worst case, we will have to give up Python 3.3 compatibility.

**Comment by hugovk** [comment]
As worst cases go, giving up Python 3.3 isn't such a bad one, it's EOL: 

https://en.wikipedia.org/wiki/CPython#Version_history

By the way, it might be an idea to also test 3.4 and 3.6 on Travis CI.

**Comment by hiranya911** [comment]
Fixed the test failure by installing `enum34` as advised.

File: integration/test_firestore.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+def test_firestore():
+    client = firestore.client()
+    expected = {
+        'name': u'Mountain View',
+        'country': u'USA',
+        'population': 77846,
+        'capital': False
+    }
+    doc = client.collection('cities').document()
```
Comment (bjornick) [inline integration/test_firestore.py]: Where are the credentials to run this stored?  Can this only be run on jenkins?

Comment (hiranya911) [reply integration/test_firestore.py]: We actually don't run integration tests on Travis. Developers like you and I could run the integration tests locally by providing credentials explicitly. See https://github.com/firebase/firebase-admin-python/blob/master/CONTRIBUTING.md#integration-testing for details.

**Comment by craigsheppard** [comment]
Is there likely to be any significant API change before this is merged in? We've been waiting on this update for two weeks now in order to release our product. I'm going to start implementing with this branch in the hopes that it is merged by the time I'm done :) Thanks for your work!

**Comment by hugovk** [comment]
I see you got Python 3.3 working. Anyway, here's the pip installs for firebase-admin from PyPI for the last month, showing very low usage for 3.3:
```
$ pypinfo --percent --pip firebase-admin pyversion
python_version percent download_count
-------------- ------- --------------
2.7              59.8%          2,549
3.6              23.9%          1,018
3.5              11.7%            501
3.4               4.0%            169
3.3               0.6%             24
2.6               0.1%              5

```

**Comment by hiranya911** [comment]
@craigsheppard The API is stable now. I'm working with various parties involved to get the release out, but I still don't have a fixed date yet. In the mean time, please do feel free to use this branch. Thanks for being patient :)

**Comment by craigsheppard** [comment]
Any idea when this will merge to master?  Thanks again!

**Comment by hiranya911** [comment]
I'm hoping next week. Fingers crossed :)

**Pull request merged**